TestDoctest.test_toolchain fails with setuptools 65.6.3

When packaging pythran for openSUSE/Factory, we have this failing. I am aware of problems with the modern setuptools, but #1984 seems to be included in 0.12.0 ([complete build log](https://github.com/serge-sans-paille/pythran/files/10088024/_log.txt) includes all packages used and steps taken to reproduce), and this still remains:

```
[ 1686s] =================================== FAILURES ===================================
[ 1686s] __________________________ TestDoctest.test_toolchain __________________________
[ 1686s] [gw3] linux -- Python 3.10.8 /usr/bin/python3.10
[ 1686s] 
[ 1686s] self = <pythran.tests.test_xdoc.TestDoctest testMethod=test_toolchain>
[ 1686s] 
[ 1686s] >   lambda self: generic_test_package(self, module))
[ 1686s] 
[ 1686s] pythran/tests/test_xdoc.py:89: 
[ 1686s] _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
[ 1686s] pythran/tests/test_xdoc.py:83: in generic_test_package
[ 1686s]     self.assertEqual(failed, 0)
[ 1686s] E   AssertionError: 1 != 0
[ 1686s] ----------------------------- Captured stdout call -----------------------------
[ 1686s] **********************************************************************
[ 1686s] File "/home/abuild/rpmbuild/BUILD/pythran-0.12.0/pythran/toolchain.py", line 443, in pythran.toolchain.compile_pythranfile
[ 1686s] Failed example:
[ 1686s]     cpp_path = compile_pythranfile('pythran_test.py', cpponly=True)
[ 1686s] Expected nothing
[ 1686s] Got:
[ 1686s]     No pythran specification, nothing will be exported
[ 1686s] **********************************************************************
[ 1686s] 1 items had failures:
[ 1686s]    1 of   7 in pythran.toolchain.compile_pythranfile
[ 1686s] ***Test Failed*** 1 failures.
[ 1686s] _____________________________ TestDoctest.test_cli _____________________________
[ 1686s] [gw6] linux -- Python 3.10.8 /usr/bin/python3.10
[ 1686s] 
[ 1686s] self = <pythran.tests.test_xdoc.TestDoctest testMethod=test_cli>
[ 1686s] 
[ 1686s]     @pytest.mark.skipif(sys.platform == "win32",
[ 1686s]                         reason="We should create a file for windows.")
[ 1686s]     def test_cli(self):
[ 1686s]         tmpfile = self.adapt_rst('../../docs/CLI.rst')
[ 1686s]         failed, _ = doctest.testfile(tmpfile, False)
[ 1686s] >       self.assertEqual(failed, 0)
[ 1686s] E       AssertionError: 1 != 0
[ 1686s] 
[ 1686s] pythran/tests/test_xdoc.py:40: AssertionError
[ 1686s] ----------------------------- Captured stdout call -----------------------------
[ 1686s] **********************************************************************
[ 1686s] File "/tmp/tmp6yedn24y", line 68, in tmp6yedn24y
[ 1686s] Failed example:
[ 1686s]     import subprocess ; res = subprocess.check_output(" /usr/bin/python3.10 -m pythran.run -e cli_foo.py -o cli_foo.hpp", shell=True).decode("ascii").strip() ; print(res, end="")
[ 1686s] Expected nothing
[ 1686s] Got:
[ 1686s]     No pythran specification, nothing will be exported
[ 1686s] **********************************************************************
[ 1686s] 1 items had failures:
[ 1686s]    1 of  25 in tmp6yedn24y
[ 1686s] ***Test Failed*** 1 failures.
[ 1686s] ----------------------------- Captured stderr call -----------------------------
[ 1686s] /usr/lib64/python3.10/runpy.py:126: RuntimeWarning: 'pythran.config' found in sys.modules after import of package 'pythran', but prior to execution of 'pythran.config'; this may result in unpredictable behaviour
[ 1686s]   warn(RuntimeWarning(msg))
```

Hmm, inside of our build systems yet another test failed as well:

```
[  413s] =================================== FAILURES ===================================
[  413s] _____________________________ TestDoctest.test_cli _____________________________
[  413s] [gw6] linux -- Python 3.8.15 /usr/bin/python3.8
[  413s] 
[  413s] self = <pythran.tests.test_xdoc.TestDoctest testMethod=test_cli>
[  413s] 
[  413s]     @pytest.mark.skipif(sys.platform == "win32",
[  413s]                         reason="We should create a file for windows.")
[  413s]     def test_cli(self):
[  413s]         tmpfile = self.adapt_rst('../../docs/CLI.rst')
[  413s]         failed, _ = doctest.testfile(tmpfile, False)
[  413s] >       self.assertEqual(failed, 0)
[  413s] E       AssertionError: 1 != 0
[  413s] 
[  413s] pythran/tests/test_xdoc.py:40: AssertionError
[  413s] ----------------------------- Captured stdout call -----------------------------
[  413s] **********************************************************************
[  413s] File "/tmp/tmpqv_r5l2d", line 68, in tmpqv_r5l2d
[  413s] Failed example:
[  413s]     import subprocess ; res = subprocess.check_output(" /usr/bin/python3.8 -m pyt
hran.run -e cli_foo.py -o cli_foo.hpp", shell=True).decode("ascii").strip() ; print(res, e
nd="")
[  413s] Expected nothing
[  413s] Got:
[  413s]     No pythran specification, nothing will be exported
[  413s] **********************************************************************
[  413s] 1 items had failures:
[  413s]    1 of  25 in tmpqv_r5l2d
[  413s] ***Test Failed*** 1 failures.
[  413s] ----------------------------- Captured stderr call -----------------------------
[  413s] /usr/lib64/python3.8/runpy.py:127: RuntimeWarning: 'pythran.config' found in sys.
modules after import of package 'pythran', but prior to execution of 'pythran.config'; thi
s may result in unpredictable behaviour
[  413s]   warn(RuntimeWarning(msg))
[  413s] =============================== warning
```

[Update build log](https://github.com/serge-sans-paille/pythran/files/10091554/_log.txt)

--- a/docs/CLI.rst
+++ b/docs/CLI.rst
@@ -64,11 +64,12 @@ That's some heavily templated code ;-) Pythran can then compile it for you to a 
 
 Pythran can also generate raw C++ code, using the ``-e`` switch::
 
-  $> printf 'msg = \"hello world\"\ndef foo(): print(msg)' > cli_foo.py
-  $> pythran -e cli_foo.py -o cli_foo.hpp
-  $> printf '#include \"cli_foo.hpp\"\nusing namespace __pythran_cli_foo ; int main() { foo()(); return 0 ; }' > cli_foo.cpp
-  $> `pythran-config --compiler --cflags` -std=c++11 cli_foo.cpp -o cli_foo
-  $> ./cli_foo
+  $> printf 'msg = \"hello world\"\ndef bar(): print(msg)' > cli_bar.py
+  $> pythran -e cli_bar.py -o cli_bar.hpp
+  No pythran specification, nothing will be exported
+  $> printf '#include \"cli_bar.hpp\"\nusing namespace __pythran_cli_bar ; int main() { bar()(); return 0 ; }' > cli_bar.cpp
+  $> `pythran-config --compiler --cflags` -std=c++11 cli_bar.cpp -o cli_bar
+  $> ./cli_bar
   hello world
 
 To know more options about Pythran, you can check::

